src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.LegacyEditor/RazorWrapperFactory.cs

```diff
@@ -75,6 +75,40 @@ internal static partial class RazorWrapperFactory
         return builder.DrainToImmutable();
     }
 
+    private static ImmutableArray<TResult> InitializeArrayWithWrappedItems<TInner, TResult>(
+        ref ImmutableArray<TResult> location,
+        IReadOnlyList<TInner> list,
+        Func<TInner, TResult> createWrapper)
+        where TInner : class
+        where TResult : class
+    {
+        var result = location;
+
+        if (result.IsDefault)
+        {
+            ImmutableInterlocked.InterlockedInitialize(ref location, WrapAll(list, createWrapper));
+        }
+
+        return result;
```

Comment:
Shouldn't this be `location`? `result` won't get updated by the `if` above.

Reply:
derp! :smile:

Reply:
(thankfully, tests failed on this bug)